src/DepthImageToLaserScan.cpp

```diff
@@ -153,7 +153,7 @@ sensor_msgs::msg::LaserScan::UniquePtr DepthImageToLaserScan::convert_msg(
 
   // Calculate and fill the ranges
   uint32_t ranges_size = depth_msg->width;
-  scan_msg->ranges.assign(ranges_size, std::numeric_limits<float>::quiet_NaN());
+  scan_msg->ranges.assign(ranges_size, std::numeric_limits<float>::infinity());
```

Comment:
This package has been publishing out-of-range values as NaN since 2012.  As such, this would represent a pretty big behavior change for it.  Can you give an example of what problems this is causing?  Also, it would be nice to get some input from @SteveMacenski on this one.

Reply:
@SteveMacenski @clalancette Thanks for taking time to look at this. 
What led me to this PR is this answer, having a similar problem with costmap not clearing: https://answers.ros.org/question/190170/local-cost-map-not-updated/
But I understand now, that it might not be such as strong answer.

Actual lidars we have used all reported inf, so I thought this was some kind industry standard.

After this discussion I went to check some lidar repos and found the following:
Using inf:
SlamTec - https://github.com/Slamtec/sllidar_ros2
Lightware - https://github.com/LightWare-Optoelectronics/lightwarelidar2
Velodyne - https://github.com/ros-drivers/velodyne
RoboSense - https://github.com/RoboSense-LiDAR/rslidar_laserscan

Using nan:
Hokuyo URG - https://github.com/ros-drivers/urg_node

Gazebo lidar module also reports infs. 

With all of the above said, I do understand this is not a strong enough case, but maybe there should be some kind of ROS defined standard for this.

Reply:
I'm OK with this change here if you can show the code for gazebo reporting INF on no-hits. 

@clalancette to be fair, it might be a decade old, but its also only really used by hobbyists and this would be a bug that few ran into and/or cared enough about to fix.

Reply:
@SteveMacenski here is a printscreen from echoing gazebo's publish to /scan:
![image](https://user-images.githubusercontent.com/108797279/221848029-75025341-af68-4210-92a1-229ab11210da.png)

Reply:
I'd like to see it in code to verify, that way we can see if that change is recent and the context behind that decision

Reply:
@SteveMacenski So sorry for late response, found it at:
https://github.com/gazebosim/gazebo-classic/blob/1bef2dc573aa15ca8a56b4a7dadb9905cf978582/gazebo/sensors/RaySensor.cc#L482

Which also references to [REF 117](https://www.ros.org/reps/rep-0117.html), which states:
```
Detections that are too close to the sensor to quantify shall be represented by -Inf. Erroneous detections shall be represented by quiet (non-signaling) NaNs. Finally, out of range detections will be represented by +Inf.
```